File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prev = 1;
        int[] res = new int[nums.length];

        for (int i = 0; i < nums.length ; i++) {
            int cur = prev;
            for (int j = i + 1 ; j < nums.length ; j++) {
                cur *= nums[j];
            }

            res[i] = cur;
            prev *= nums[i];
            cur = 0;
        }
        
        return res;
    }
}  

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        #Lookin at this we need to determin all subsets
        # this means for 1, 2
            # [] [1] [2] [1,2]

        # list of lists 


        res = []
        cur = []
        def dfs(i):
            if i >= len(nums):
                res.append(cur[:])
                return
            cur.append(nums[i])
            dfs(i+1)
            cur.pop()
            dfs(i+1)

        dfs(0)
        return res

